pymongo/message.py

```diff
@@ -431,8 +441,12 @@ def get_message(self, dummy0, sock_info, use_cmd=False):
         if use_cmd:
             spec = self.as_command(sock_info)[0]
             if sock_info.op_msg_enabled:
+                if self.exhaust_mgr:
```

Comment:
So I understand. The intended flow is roughly:
- Create Cursor() with exhaust option.
- run first find command with OP_MSG
- Cursor receives a ExhaustResponse and initializes Cursor.exhaust_mgr, pinning a connection
- Run the next getMore: we set the exhaustAllowed flag.
- the server responses with the nextBatch with the moreToCome bit set.

Reply:
Yes. And on the subsequent `getMore`, we don't send anything, we simply see that `more_to_come` is set on the exhaust_mgr and so we start receiving on the socket pinned to the manager.

Reply:
If anything, it might be better to move the test from TestCollection to here?